lib/constants.js

```diff
@@ -0,0 +1,6 @@
+// this is to prevent overly long strings
+const maxLength = 100;
```

Comment:
This naming is vague.  Maybe qualified naming like "maxFooLength" for common maximums would help?  That could at least remove the oddness of "maxLength * 2" where the maxLength isn't even a max!  There are lots of varied uses below, so I'm not sure what the best pattern is here.

Reply:
This is general one of the problems with this PR. I have tried to removed maxLength where I could, but there are still some there.